ftw/news/browser/news_listing_block.py

```diff
@@ -49,9 +50,15 @@ def get_block_info(self):
         return info
 
     def get_news(self):
+        show_inactive = (not api.user.is_anonymous()) and api.user.has_permission(
+            'ftw.news: Add News',
+            username=api.user.get_current().getId(),
+            obj=api.portal.get(),
```

Comment:
@jone, this will not work for local roles. The problem is, that the query may get news items from various containers if configured to do so. We would then check the permission on every container or make multiple queries and join the results. What do you think?